internal/proxy/providers/sso_test.go

```diff
@@ -230,6 +240,24 @@ func TestSSOProviderGetEmailAddress(t *testing.T) {
 				Groups: []string{"core@gsa.gov"},
 			},
 		},
+		{
+			Name: "redeem-internal successful",
+			Code: "code1234",
+			RedeemResponse: &redeemResponse{
+				AccessToken:  "a1234",
+				ExpiresIn:    10,
+				RefreshToken: "refresh12345",
+				Email:        "michael.bland@gsa.gov",
+			},
+			ProxyProviderURL: &url.URL{
```

Comment:
I think what you'd want to do to test this behavior is create a `ProxyProviderResponse` similar to the Redeem Response and Profile Response, as well as a `ProxyRedeemResponse`, and assign the server urls to those provider fields. Then you can test that the endpoints are being hit based on the response.